src/components/cdTab.js

```diff
@@ -1,34 +1,183 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-import Grid from '@mui/material/Grid'
-import { Skeleton } from '@mui/material'
+import {
+  Alert,
+  Box,
+  FormControlLabel,
+  FormGroup,
+  Switch,
+  Tooltip,
+} from '@mui/material'
 
 import ObjectTextViewer from './util/ObjectTextViewer'
+import { useFetchComponentDescriptor } from '../api/useFetch'
+import CenteredSpinner from './util/CenteredSpinner'
+
+
+const FetchRawComponent = ({
```

Comment:
Why modelling fetching of the raw component descriptor as a standalone React component? I think this isn't very intuitive especially because there is no return value. Maybe you can achieve the same conditional semantics using an optional wrapper component of `Viewer` and `Options` which takes care of fetching the raw component descriptor?